### DISCUSSION TITLE
Terminal Context Menu item group names

### DISCUSSION THREAD

**Discussion by Pranith12250** [microsoft/vscode-discussions#1082] [Extension Development QnA] [answered]
I am trying to develop an extension that shows a context menu item in the terminal. So, for this, I have the following in the package.json file:
```
"menus": 
    {
      "terminal/context":
      [
        {
          "command": "myextension.solveError",
          "group":"somegroup"
        }
      ]
    }
```

with the command being:

```
"commands": 
[
      {
        "command": "myextension.solveError",
        "category": "MyExtension",
        "title": "Resolve error"
      }
]
```

Now this works correctly as expected:
<img width="306" alt="Screenshot 2024-03-05 at 2 05 38 AM" src="https://github.com/microsoft/vscode-discussions/assets/127399614/7c3bda3b-90a4-46d1-a7be-ece020455ae0">

But, The menu item is appearing at the very bottom. Suppose, I want the item to appear in the group containing "New Terminal" and "Split Terminal" or the group containing "Copy", "Copy as HTML" and all. For this, I would have to change the group name from the earlier code in:

`"group":"somegroup"`

But what would the name of the group be? I tried looking for documentation for this but could not find any for terminal context. I found group names for editor context in the [vscode API Docs](https://code.visualstudio.com/api/references/contribution-points#Sorting-of-groups:~:text=%5D%0A%20%20%7D%0A%7D-,Sorting%20of%20groups,-Menu%20items%20can) but could not find anything for terminal context. For example, from the docs, for editor context, the group names are "navigation", "1_modification" and so on. What are the similar group names for terminal context?

**Comment by alefragnani** [comment, marked as answer]
Hi @Pranith12250 ,

I guess you are looking for these values:

```ts
const enum ContextMenuGroup {
	Create = '1_create',
	Edit = '2_edit',
	Clear = '3_clear',
	Kill = '4_kill',
	Config = '5_config'
}
```

Which are defined here https://github.com/microsoft/vscode/blob/191be39e5ac872e03f9d79cc859d9917f40ad935/src/vs/workbench/contrib/terminal/browser/terminalMenus.ts#L21.

The info seems to be missing in the API Docs, as you pointed out. I think worth opening an issue in the docs repo https://github.com/microsoft/vscode-docs, so other extension authors would find the info easier.

Hope this helps

**Reply by Pranith12250** [reply]
Hey thanks a lot! That is exactly what I was looking for. I will do that.
